extensions/notebook/src/book/bookModel.ts

```diff
@@ -32,10 +32,29 @@ export class BookModel {
 		public readonly openAsUntitled: boolean,
 		public readonly isNotebook: boolean,
 		private _extensionContext: vscode.ExtensionContext,
+		private _bookTreeViewEvent: vscode.EventEmitter<BookTreeItem | undefined>,
 		public readonly notebookRootPath?: string) {
 		this._bookItems = [];
 	}
 
+	public unwatchTOC(): void {
+		fs.unwatchFile(this.tableOfContentsPath);
+	}
+
+	public watchTOC(): void {
+		fs.watchFile(this.tableOfContentsPath, async (curr, prev) => {
+			if (curr.mtime > prev.mtime) {
+				await this.reInitializeContents();
+			}
+		});
+	}
+
+	@debounce(1500)
+	public async reInitializeContents(): Promise<void> {
+		await this.initializeContents();
```

Comment:
Do you also want to handle concurrent executions? If this is called multiple times outside of the debounce window then it'll call initialize contents multiple times which could have weird side effects depending on which one finished first.

Reply:
Don't previous pending calls get cancelled when using a debounce?

Reply:
I'm saying ones outside the debounce window.

- Call comes in - debounce timer starts.
- Wait 1.5 seconds
- initializeContents() is called (timer expired) and begins to run
- reInitializeContents is called again - timer starts again
- Wait 1.5 seconds
- initializeContents() is called again

We now have initializeContents running twice separately. If the call takes >1.5seconds to complete this is a scenario that I could easily see happening.

Reply:
mm I didn't consider that, do you think it'll be better to add the debounce decorator directly to the initalizeContents method, instead?

Reply:
That doesn't really solve the problem. And in general this isn't something that can be solved with debouncing.

You'd need to add some sort of gating to ensure that only one refresh is running at a time and/or additional calls are queued up for execution instead of executing right away. We can chat offline if you aren't sure where to start for that.